vault/resource_mongodbatlas_secret_role.go

```diff
@@ -124,7 +124,7 @@ func mongodbAtlasSecretRoleCreateUpdate(ctx context.Context, d *schema.ResourceD
 		consts.FieldMaxTTL,
 	}
 	for _, k := range fields {
-		if d.HasChange(k) {
+		if d.HasChange(k) || k == consts.FieldRoles {
 			data[k] = d.Get(k)
```

Comment:
I suggest making this a bit more future-proof:
```suggestion
		if v, ok := d.GetOk(k); ok {
			data[k] = v
```

Reply:
Thanks, agree.